### PR TITLE
feat(#19183): make Chronicle namespace a templateable field

### PR DESCRIPTION
**Pull request by ChocPanda**

### CODE REVIEW

**Comment by ChocPanda** [comment]
Hey @StephenWakely ,
I appreciated your help with https://github.com/vectordotdev/vector/pull/19398 and thought I'd try making the namespace a templateable field, as you suggested.
Would you mind taking a look, it feels like there should have been a simpler solution that didn't require making my own sink and partitioner

**Review by brett0000FF — APPROVED** [review]

**Review by StephenWakely — APPROVED** [review]
Looks good to me. A tiny niggle, but that's not really been introduced by this PR, so it's fine to go!

File: changelog.d/19183_templateable_chronicle_namespace.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Make the Chronicle namespace a templatable field
```
Comment (StephenWakely) [inline changelog.d/19183_templateable_chronicle_namespace.md:1]: This changelog needs to be named `changelog.d/19183_templateable_chronicle_namespace.enhancement.md`.

**Review by StephenWakely — CHANGES_REQUESTED** [review]
Can you run `chronicled_unstructured.rs` through `rustfmt`.

**Review by StephenWakely — APPROVED** [review]

**Pull request merged**